File: Java Learning/JavaCode/src/Day_05/BubbleSort.java
package Day_05;

/**
 * 冒泡排序
 * 将[24,69,80,57,13]五个数升序排列
 */
public class BubbleSort {
    public static void main(String[] args) {
        //化繁为简，先死后活
        int[] arr = {2,2,4,4,2,2,4,4};
        int temp;
        int n = arr.length;//将数组的长度赋值给一个变量。可以随意添加数组的元素
        for (int j = 0; j < n-1; j++) {
            for (int i = 0; i < n-1-j; i++) {
                if (arr[i] > arr[i + 1]) {
                    temp = arr[i];
                    arr[i] = arr[i + 1];
                    arr[i + 1] = temp;
                }
            }
            System.out.print("第"+(j+1)+"轮排序后arr为：");
            for(int i = 0; i < arr.length; i++) {
                System.out.print(arr[i]+" ");
            }
            System.out.println();
        }
    }
}

File: C++Learning/C++basisCode/day05-array/BubbleSort.cpp
//
// Created by szy on 2021/10/21.
//冒泡排序

#include <iostream>

using namespace std;

//冒泡排序
int main() {
    int arr[] = {32, 453, 3, 0, -4, 56, 2};
    int arrLength = sizeof(arr) / sizeof(arr[0]); //获取数组长度
    cout << "原数组: arr=[";
//格式化数组
    for (int i = 0; i < arrLength; i++) {
        if (i == arrLength - 1) {
            cout << arr[i] << "]";
        } else {
            cout << arr[i] << ",";
        }
    }

    int temp = 0;
    //排序过程
    for (int i = 0; i < arrLength - 1; i++) {
        for (int j = 0; j < arrLength - 1 - i; j++) {
            if (arr[j] > arr[j + 1]) {
                temp = arr[j];
                arr[j] = arr[j + 1];
                arr[j + 1] = temp;
            }
        }
    }
    cout << "\n冒泡排序后的数组: newArr=[";
    for (int i = 0; i < arrLength; i++) {
        if (i == arrLength - 1) {
            cout << arr[i] << "]";
        } else {
            cout << arr[i] << ",";
        }
    }
    system("pause");
    return 0;
}